src/App.tsx

```diff
@@ -1,93 +1,142 @@
+import { Footer } from './components/Footer';
+
+const USER_ID = 7075;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isEnterPressed, setIsEnterPressed] = useState(false);
+  const [selectFilter, setSelectFilter] = useState('all');
+  const [allTodosCompleted, setAllTodosCompleted] = useState<boolean>(true);
+
+  const count = todos.length;
+
+  useEffect(() => {
+    getTodos(USER_ID).then(todosFromServer => {
+      setTodos(todosFromServer);
+    });
+  }, [isEnterPressed,
+  ]);
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter((todo: Todo) => {
+      switch (selectFilter) {
+        case 'active':
+          return !todo.completed;
+        case 'completed':
+          return todo.completed;
+        default:
+          return todos;
+      }
+    });
+  }, [
+    todos,
+    selectFilter,
+  ]);
+
+  if (!USER_ID) {
+    return <UserWarning />;
+  }
+
+  const handleDeleteTodo = (todoId: number) => {
+    deleteTodo(todoId);
+    setTodos(todos.filter(todo => todo.id !== todoId));
+  };
+
+  const handleChangeTodo = async (
+    todoId: number,
+    updatedFields: { title?: string; completed?: boolean },
+  ) => {
+    await updateTodo(todoId, updatedFields);
+
+    setTodos(state => state.map(todo => {
+      if (todo.id === todoId) {
+        return { ...todo, ...updatedFields };
+      }
+
+      return todo;
+    }));
+  };
+
+  const clearCompleted = () => {
+    todos
+      .filter(todo => todo.completed)
+      .forEach(async todo => {
+        await deleteTodo(todo.id);
+        setTodos(prevTodos => prevTodos
+          .filter(prevTodo => prevTodo.id !== todo.id));
+      });
+  };
+
+  const handleToggleAll = () => {
+    const updatedTodos = todos.map(todo => (
+      {
+        ...todo,
+        completed: allTodosCompleted,
+      }
+    ));
+
+    setTodos(updatedTodos);
+    setAllTodosCompleted(prevCompleted => !prevCompleted);
+  };
+
+  const isCompleted = todos.some(todo => todo.completed === true);
+
   return (
     <div className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
-
-        <form>
-          <input
-            type="text"
-            data-cy="createTodo"
-            className="new-todo"
-            placeholder="What needs to be done?"
-          />
-        </form>
-      </header>
-
-      <section className="main">
-        <input
-          type="checkbox"
-          id="toggle-all"
-          className="toggle-all"
-          data-cy="toggleAll"
+      <Header
+        inputText={todoTitle}
+        setInputText={(ev) => {
+          setTodoTitle(ev.target.value);
+          setIsEnterPressed(false);
+        }}
+        handleKeyPress={(event) => {
+          if (event.key === 'Enter') {
+            event.preventDefault();
+            createTodo(7075, todoTitle);
+          }
+        }}
+        handleKeyUp={(event) => {
```

Comment:
it is bad practise to use anonymous arrow function in jsx.
make handler function and pass reference here
please check and fix all similar cases